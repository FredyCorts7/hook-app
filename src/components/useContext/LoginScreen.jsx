import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const newUser = {
    id: 1234,
    name: 'Camila',
  };

  return (
    <div>
      <h1>Login Screen</h1>
      <hr />

      <button className='btn btn-outline-info' onClick={() => setUser(newUser)}>
        Log In
      </button>
    </div>
  );
};

export default LoginScreen;
