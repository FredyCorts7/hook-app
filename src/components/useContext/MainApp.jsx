import React from 'react';
import AppRouter from './AppRouter';
import { UserContext } from './UserContext';

const MainApp = () => {
  const user = {
    id: 123,
    name: 'Fredy C',
    email: 'fred.cor.14@gmail.com',
  };

  return (
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
