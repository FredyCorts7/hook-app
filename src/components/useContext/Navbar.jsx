import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  return (
    <nav className='navbar navbar-expand-sm navbar-dark bg-dark'>
      <NavLink to='/' className='navbar-brand'>
        Navbar
      </NavLink>
      <div className='collapse navbar-collapse' id='navbarNav'>
        <ul className='navbar-nav'>
          <li className='nav-item'>
            <NavLink exact activeClassName='active' to='/' className='nav-link'>
              Home
            </NavLink>
          </li>
          <li className='nav-item'>
            <NavLink
              exact
              activeClassName='active'
              to='/about'
              className='nav-link'
            >
              About
            </NavLink>
          </li>
          <li className='nav-item'>
            <NavLink
              exact
              activeClassName='active'
              to='/login'
              className='nav-link'
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
