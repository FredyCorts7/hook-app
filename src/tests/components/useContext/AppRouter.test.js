import React from 'react';
import { mount } from 'enzyme';
import AppRouter from '../../../components/useContext/AppRouter';
import { UserContext } from '../../../components/useContext/UserContext';
import '@testing-library/jest-dom';

describe('Testing to <AppRouter />', () => {
  const user = {
    name: 'Maria',
    email: 'mari@gmail.com',
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );

  test('should show succesfully', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
