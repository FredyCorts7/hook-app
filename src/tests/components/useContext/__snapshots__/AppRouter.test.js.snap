// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <AppRouter /> should show succesfully 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Navbar
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "name": "Maria",
   "email": "mari@gmail.com"
}
      </pre>
    </div>
  </div>
</div>
`;
